mmv1/products/datastream/product.yaml

```diff
@@ -17,6 +17,9 @@ versions:
   - !ruby/object:Api::Product::Version
     name: ga
     base_url: https://datastream.googleapis.com/v1/
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://datastream.googleapis.com/v1/
```

Comment:
Why is this necessary? I would expect MMv1 to fall back to the GA URL if beta isn't provided.

Reply:
Without this I get the following error when trying to build the beta provider with the new sql server configs set to `min_version: beta`

```sh
bundler: failed to load command: compiler.rb (compiler.rb)
/Users/nathan.bugden/github/nbugden/magic-modules/mmv1/api/product.rb:119:in `version_obj': API version 'beta' does not exist for product 'Datastream' (RuntimeError)
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/api/type.rb:423:in `min_version'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/api/type.rb:435:in `exclude_if_not_in_version!'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/api/type.rb:717:in `exclude_if_not_in_version!'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/api/resource.rb:586:in `block in exclude_if_not_in_version!'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/api/resource.rb:586:in `each'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/api/resource.rb:586:in `exclude_if_not_in_version!'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/provider/terraform.rb:372:in `block in generate_objects'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/provider/terraform.rb:361:in `each'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/provider/terraform.rb:361:in `generate_objects'
        from /Users/nathan.bugden/github/nbugden/magic-modules/mmv1/provider/terraform.rb:92:in `generate'
        from compiler.rb:267:in `block in <top (required)>'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:587:in `call_with_index'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:557:in `process_incoming_jobs'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:537:in `block in worker'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:528:in `fork'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:528:in `worker'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:519:in `block in create_workers'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:518:in `each'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:518:in `each_with_index'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:518:in `create_workers'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:457:in `work_in_processes'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/parallel-1.22.1/lib/parallel.rb:294:in `map'
        from compiler.rb:142:in `<top (required)>'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/cli/exec.rb:58:in `load'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/cli/exec.rb:58:in `kernel_load'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/cli/exec.rb:23:in `run'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/cli.rb:491:in `exec'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/cli.rb:34:in `dispatch'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/cli.rb:28:in `start'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/exe/bundle:45:in `block in <top (required)>'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.4.6/exe/bundle:33:in `<top (required)>'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/bin/bundle:25:in `load'
        from /Users/nathan.bugden/.rbenv/versions/3.1.0/bin/bundle:25:in `<main>'
make[1]: *** [mmv1] Error 1
make: *** [provider] Error 2
```
If there is a preferred way to solve this that doesn't include adding the beta provider let me know. I'd be happy to make the change.

Reply:
@melinath can I resolve this? Or do you have another question/suggestion?

Reply:
This resolution is fine, I'm just surprised it's necessary.

Reply:
Yeah I was surprised as well